src/main.cpp

```diff
@@ -2553,6 +2553,7 @@ enum FlushStateMode {
  * or always and in all cases if we're in prune mode and are deleting files.
  */
 bool static FlushStateToDisk(CValidationState &state, FlushStateMode mode) {
+    size_t cacheSize = mempool.DynamicMemoryUsage();
```

Comment:
I wonder if it would be better to call `LimitMempoolSize` right before this?   During a reorg the mempool size isn't limited until the end, so theoretically `FlushStateToDisk` could get called and have its effective cache size reduced below the expected minimum.  @sdaftuar Is there any harm in calling an extra `LimitMempoolSize` in the middle of a reorg other than possible inefficiencies of what you're evicting?